pkg/provider/aci.go

```diff
@@ -960,68 +968,68 @@ func (p *ACIProvider) getInitContainers(ctx context.Context, pod *v1.Pod) ([]aza
+			Properties: &azaci.InitContainerPropertiesDefinition{
 				Image:                &pod.Spec.InitContainers[i].Image,
-				Command:              p.getCommand(&pod.Spec.InitContainers[i]),
-				VolumeMounts:         p.getVolumeMounts(&pod.Spec.InitContainers[i]),
-				EnvironmentVariables: p.getEnvironmentVariables(&pod.Spec.InitContainers[i]),
+				Command:              p.getCommand(pod.Spec.InitContainers[i]),
+				VolumeMounts:         p.getVolumeMounts(pod.Spec.InitContainers[i]),
+				EnvironmentVariables: p.getEnvironmentVariables(pod.Spec.InitContainers[i]),
 			},
 		}
 
-		initContainers = append(initContainers, newInitContainer)
+		initContainers = append(initContainers, &newInitContainer)
 	}
 	return initContainers, nil
 }
 
-func (p *ACIProvider) getContainers(pod *v1.Pod) (*[]azaci.Container, error) {
-	containers := make([]azaci.Container, 0, len(pod.Spec.Containers))
+func (p *ACIProvider) getContainers(pod *v1.Pod) ([]*azaci.Container, error) {
+	containers := make([]*azaci.Container, 0, len(pod.Spec.Containers))
 
 	podContainers := pod.Spec.Containers
 	for c := range podContainers {
 
 		if len(podContainers[c].Command) == 0 && len(podContainers[c].Args) > 0 {
 			return nil, errdefs.InvalidInput("ACI does not support providing args without specifying the command. Please supply both command and args to the pod spec.")
 		}
-		cmd := append(podContainers[c].Command, podContainers[c].Args...)
-		ports := make([]azaci.ContainerPort, 0, len(podContainers[c].Ports))
+		cmd := p.getCommand(podContainers[c])
+		ports := make([]*azaci.ContainerPort, 0, len(podContainers[c].Ports))
 		aciContainer := azaci.Container{
 			Name: &podContainers[c].Name,
-			ContainerProperties: &azaci.ContainerProperties{
+			Properties: &azaci.ContainerProperties{
 				Image:   &podContainers[c].Image,
-				Command: &cmd,
-				Ports:   &ports,
+				Command: cmd,
+				Ports:   ports,
 			},
 		}
 
 		for i := range podContainers[c].Ports {
-			containerPorts := aciContainer.Ports
-			containerPortsList := append(*containerPorts, azaci.ContainerPort{
+			containerPorts := aciContainer.Properties.Ports
+			containerPortsList := append(containerPorts, &azaci.ContainerPort{
 				Port:     &podContainers[c].Ports[i].ContainerPort,
 				Protocol: util.GetProtocol(podContainers[c].Ports[i].Protocol),
 			})
-			aciContainer.Ports = &containerPortsList
+			aciContainer.Properties.Ports = containerPortsList
 		}
 
-		volMount := make([]azaci.VolumeMount, 0, len(podContainers[c].VolumeMounts))
-		aciContainer.VolumeMounts = &volMount
+		volMount := make([]*azaci.VolumeMount, 0, len(podContainers[c].VolumeMounts))
+		aciContainer.Properties.VolumeMounts = volMount
 		for v := range podContainers[c].VolumeMounts {
-			vol := aciContainer.VolumeMounts
-			volList := append(*vol, azaci.VolumeMount{
+			vol := aciContainer.Properties.VolumeMounts
+			volList := append(vol, &azaci.VolumeMount{
```

Comment:
Can you directly append aciContainer.Properties.VolumeMounts?

Reply:
Yes, this was a side effect of the SDK v1 pointer issue. Removed.